src/main.rs

```diff
@@ -1,3 +1,156 @@
-fn main() {
-    println!("Hello, World!");
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * SPDX-FileCopyrightText: Copyright 2023 KUNBUS GmbH
+ */
+
+//! Test video and audio output by showing coloured frames on the framebuffer and playing back a
+//! siren over the default ALSA audio device.
+
+use alsa::pcm::{Access, Format, HwParams, State, PCM};
+use alsa::{Direction, ValueOr};
+use anyhow::{bail, Context};
+use framebuffer::{Framebuffer, KdMode};
+use std::{thread, time};
+
+const TTY: &str = "/dev/tty1";
+const FB: &str = "/dev/fb0";
+
+// how long a frame is displayed in seconds
+const FRAME_LENGTH: u64 = 1;
+// amount of time one audio sample should play
+const AUDIO_LENGTH: u64 = 1;
+
+fn frame_set_color(frame: &mut [u8], color: (u8, u8, u8)) {
+    for pixel in &mut frame.chunks_mut(4) {
+        pixel[0] = color.2;
+        pixel[1] = color.1;
+        pixel[2] = color.0;
+        pixel[3] = 0xff;
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    if let Some(a) = std::env::args().nth(1) {
+        if a == "--version" {
```

Comment:
```suggestion
    if let Some(arg) = std::env::args().nth(1) {
        if arg == "--version" {
```
Why not using `arg` as name. It is not much longer and it caries all the information. I like concise but clear identifiers. And I don't think these are against idiomatic rust.